### PR TITLE
fix bug with passing arguments to criterion callbacks in key-value fo…

### PR DESCRIPTION
**Pull request by Inkln**
…rmat

## Description

When i use loss like this
```
loss_wh:
  callback: CriterionCallback
  input_key: null
  output_key: null
  prefix: loss_wh
  criterion_key: l1_wh
  multiplier: *wh_weight
```
In dl/callbacks/criterion.py method _compute_loss_key_value fails
```
def _compute_loss_key_value(self, state: RunnerState, criterion):
        output = self._get_output(state.output, self.output_key)
        input = self._get_input(state.input, self.input_key)
        loss = criterion(**output, **input)
        return loss
```
Because there are same keys in ```**input``` and ```**output```

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves an existing feature)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have read the [Code of Conduct](https://github.com/catalyst-team/catalyst/blob/master/CODE_OF_CONDUCT.md) document.
- [x] I have read the [Contributing](https://github.com/catalyst-team/catalyst/blob/master/CONTRIBUTING.md) guide.
- [ ] I have checked the code-style using `make check-style`.
- [ ] I have written the docstring in Google format for all the methods and classes that I used.
- [x] I have checked the docs using `make check-docs`.

### CODE REVIEW

**Comment by Scitator** [comment]
Sorry, but I can argue, that for clear implementation it's better not to use save KV for input and target. So such issue was expected :)

**Comment by Scitator** [comment]
Discussed internally :)
